airflow/upgrade/rules/airflow_macro_plugin_removed.py

```diff
@@ -39,9 +39,16 @@ def _check_file(self, file_path):
         problems = []
         class_name_to_check = self.MACRO_PLUGIN_CLASS.split(".")[-1]
         with open(file_path, "r") as file_pointer:
-            for line_number, line in enumerate(file_pointer, 1):
-                if class_name_to_check in line:
-                    problems.append(self._change_info(file_path, line_number))
+            try:
+                for line_number, line in enumerate(file_pointer, 1):
+                    if class_name_to_check in line:
+                        problems.append(self._change_info(file_path, line_number))
+            except UnicodeDecodeError:
+                raise Exception(
+                    "Unable to read file {}, it may need to be added to the .airflowignore".format(
```

Comment:
Imho we should add this info to `problem` instead of rising the exception. In this way the command will work and users will be informed about all problems - including undesired files. WDYT?

Reply:
More than happy to add this to the problems report! My gut instinct for big projects like this is "preserve existing behavior by default", so I didn't want to change things up too much initially 😄  Also good to know on the macros! I can add that to the logic as well. Can they be python files that don't have the `.py` extension but are ostensibly python otherwise?

Reply:
Plugins can only be python files with `.py` extension, otherwise we skip them:
https://github.com/apache/airflow/blob/748d05f519c0e5427d2a46763591f03697c4d585/airflow/plugins_manager.py#L200-L201